megamek/src/megamek/common/util/TurnTimer.java

```diff
@@ -0,0 +1,119 @@
+package megamek.common.util;
+
+import megamek.client.Client;
+import megamek.client.ui.swing.AbstractPhaseDisplay;
+import megamek.common.IGame;
+import megamek.common.options.Option;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+/**
+ * This class takes a time limit, which is to be set in Basic Options and counts down to zero
+ * When zero is reached, the ready() method of the given {@link AbstractPhaseDisplay} is called
+ * to end the users current turn.
+ *
+ */
+public class TurnTimer {
+
+    private Timer timer;
+    private JProgressBar progressBar;
+    private ActionListener listener;
+    private JLabel remaining;
+    private JDialog display;
+    private int timeLimit ;
+
+
+    public TurnTimer(int limit, AbstractPhaseDisplay phaseDisplay) {
+
+        timeLimit = limit;
+        display = new JDialog();
+        progressBar = new JProgressBar(JProgressBar.VERTICAL, 0, limit);
+        progressBar.setValue(limit);
+        progressBar.setString(String.valueOf(timeLimit)+"s");
+
+        listener = new ActionListener() {
+            int counter = limit;
+            public void actionPerformed(ActionEvent ae) {
+                counter--;
+                remaining.setText(counter+" sec");
+                progressBar.setValue(counter);
+                progressBar.setString(String.valueOf(counter)+"s");
```

Comment:
Redundant call to String#valueOf in concatenation.
```suggestion
                progressBar.setString(counter + "s");
```